src/v/cluster/tests/serialization_rt_test.cc

```diff
@@ -2366,13 +2371,19 @@ static void check_async_serde_no_forgotten_fields() {
         << ss::pretty_type_name(typeid(T)));
 };
 
+template<typename T>
+static void check_async_serde_no_forgotten_fields() {}
+
 SEASTAR_THREAD_TEST_CASE(test_controller_snapshot_serde_no_forgotten_fields) {
     // Check for every controller snapshot part with manual async serialization
     check_async_serde_no_forgotten_fields<cluster::controller_snapshot>();
     check_async_serde_no_forgotten_fields<
       cluster::controller_snapshot_parts::topics_t>();
     check_async_serde_no_forgotten_fields<
       cluster::controller_snapshot_parts::topics_t::topic_t>();
+
+    static_assert(std::is_standard_layout_v<
```

Comment:
Hmm does this mean that we are effectively only checking `security_t`? Then the test doesn't look so useful anymore (looking at the evolution of snapshot parts, the only structs that have been modified since the snapshot was introduced 9 months ago are: the snapshot itself and topics and those are precisely those that are not checked). So I would be in favor of removing the test completely.

Reply:
fixed